### PR TITLE
refactor: use contains() to improve readability

### PR DESCRIPTION
**Pull request by zielint0**
- use contains() to clearly say intentions (both commits)
- simplify method *canBeRegExpPattern* (one commit)

### CODE REVIEW

File: src/main/java/spoon/reflect/path/impl/CtNamedPathElement.java [hunk 1]
```diff
@@ -62,12 +62,9 @@ public CtNamedPathElement(String pattern, boolean canBeRegexp) {
 	}
 
 	private boolean canBeRegExpPattern(String str) {
-		if (str.indexOf("()") >= 0) {
-			//this is acceptable by RegExp, but is in conflict with method signature
-			//and it makes no sense for regexp
-			return false;
-		}
-		return true;
+		//this is acceptable by RegExp, but is in conflict with method signature
```
Comment (monperrus) [inline src/main/java/spoon/reflect/path/impl/CtNamedPathElement.java]: here the meaning of the comment is less clear. The previous version is better.

File: src/main/java/spoon/reflect/path/CtPathStringBuilder.java [hunk 2]
```diff
@@ -88,7 +88,7 @@ public CtPath fromString(String pathStr) throws CtPathException {
 			String kind = token;
 			CtPathElement pathElement = null;
 			token = tokenizer.getNextToken(PATH_DELIMITERS);
-			if (token != null && token.length() == 1 && PATH_DELIMITERS.indexOf(token) >= 0) {
+			if (token != null && token.length() == 1 && PATH_DELIMITERS.contains(token)) {
```
Comment (monperrus) [inline src/main/java/spoon/reflect/path/CtPathStringBuilder.java]: ok for this one.

File: src/main/java/spoon/reflect/path/impl/CtNamedPathElement.java (hunk 1 above)
Comment (zielint0) [reply src/main/java/spoon/reflect/path/impl/CtNamedPathElement.java]: How about this one?
```
	private boolean canBeRegExpPattern(String str) {
		//this is acceptable by RegExp, but is in conflict with method signature
		//and it makes no sense for regexp
		return str.indexOf("()") < 0;
	}
```

**Comment by monperrus** [comment]
updated comment and merged.

**Pull request merged**